Do not show 'work link' if work transfer is cancelled'

njaffer opened the pull request:
related to #1552

jcoyne commented on app/helpers/sufia/sufia_helper_behavior.rb:
```diff
@@ -31,7 +31,7 @@ def error_messages_for(object)
 
     def show_transfer_request_title(req)
-      if req.deleted_work?
+      if req.deleted_work? || req.status != 'Canceled'
```
Have you considered making this a method on the `req` object?

njaffer replied:
Sure, I will update the code

jcoyne commented on app/models/proxy_deposit_request.rb:
```diff
@@ -75,6 +75,10 @@ def cancel!
   end
 
+  def is_canceled?
+    self.status == 'canceled'
```
Let's make `'canceled'` a constant (and use it in `cancel!`) too.  Also, can you add a test to this PR?

jcoyne commented on app/models/proxy_deposit_request.rb:
```diff
@@ -75,6 +75,10 @@ def cancel!
     save!
   end
 
+  def is_canceled?
```
The ruby convention is just `canceled?` without having `is_` at the front.

njaffer commented:
@jcoyne: The test already covers the canceled status here: https://github.com/projecthydra/sufia/blob/master/spec/models/proxy_deposit_request_spec.rb#L63

and for making 'canceled' as a constant, we will have to create all other status as constants also, how about we make that as a separate ticket?

jcoyne commented:
There is no test of the behavior of the new method or how it affects the view.

njaffer commented:
@jcoyne: Is there an existing test that needs to be modified? I cannot seem to find where would this test go?

jcoyne commented:
@njaffer I'd put one test in here: https://github.com/njaffer/sufia/blob/transfer/spec/models/proxy_deposit_request_spec.rb  and then one in here: https://github.com/projecthydra/sufia/blob/master/spec/helpers/sufia_helper_spec.rb  I did mistype before, I said view, but I actually meant helper.

awead commented:
:ok_hand: to me, but @jcoyne might want a test in `sufia_helper_spec.rb`.

njaffer commented:
@awead , @jcoyne : I do not see any relevant test for testing the link for work name in sufia_helper_spec.rb 

The test I added in proxy_deposit_request_spec.rb tests the code I have added in the proxy_deposit_request file

jcoyne commented:
@njaffer Do you think it might be important to test the behavior of show_transfer_request_title when the status of the request is canceled? If you don't add a test here your change might be reverted without anyone knowing this was important behavior.

njaffer commented:
@jcoyne: added the test to make sure no one can put title as a link when the transferred work is cancelled!
